source/developer/contribution-guide.md

```diff
@@ -20,41 +20,22 @@ Once you have a ticket:
 
 1. Follow [developer setup instructions](https://github.com/mattermost/platform/blob/master/doc/developer/Setup.md) to install Mattermost
 
-2. Create a branch with <branch name> set to the ID of the ticket you're working on, for example ```PLT-394```, using command: 
+2. Create a branch with `<branch name>` set to the ID of the ticket you're working on, for example `PLT-394`, using command: `git checkout -b <branch name>`
 
-```
-git checkout -b <branch name>
-```
+## Preparing a Pull Request 
 
-## Programming and Testing 
-
-1. Please review the [Mattermost Style Guide](doc/developer/Style-Guide.md) prior to making changes
-
-   To keep code clean and well structured, Mattermost uses ESLint to check that pull requests adhere to style guidelines for React. In addition all code is run through the official go formatter tool gofmt. Code will need to follow Mattermost's React style guidelines and the golang official style guide in order to pass the automated build tests when a pull request is submitted. 
-   
-2. Please make sure to thoroughly test your change before submitting a pull request
-
-   For any changes to text processing, please run the text processing tests found in the [/tests](https://github.com/mattermost/platform/tree/master/doc/developer/tests) folder in GitHub.
-   
-   Please review the ["Fast, Obvious, Forgiving" experience design principles](http://www.mattermost.org/design-principles/) for Mattermost and check that your feature meets the criteria. Also, for any changes to user interface or help text, please read the changes out loud, as a quick and easy way to catch any inconsitencies
-
-3. For new server-side funcitonality, please include test cases that verify the code performs as you have intended
+Before submitting a pull request (PR), check that:  
 
+1. You’ve signed the [Contributor License Agreement](http://www.mattermost.org/mattermost-contributor-agreement/), so you can be added 2. Your change has an [APR Ticket](http://docs.mattermost.com/process/accepting-pull-request.html)
+to the Mattermost [Approved Contributor List](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true)  
+3. Your code follows the [Mattermost Style Guide](http://docs.mattermost.com/developer/style-guide.html)  
+4. Unit tests are included for new server side functionality 
+5. Localization support is included  
+6. Change meets UX Guidelines of [Fast, Obvious, Forgiving](http://www.mattermost.org/design-principles/)
+7. If changes involves documentation, PR is submitted to [mattermost/docs](https://github.com/mattermost/docs)  
```

Comment:
Fix spelling (changes involves), also maybe make it clearer when a change needs documentation